integrations/anthropic/src/haystack_integrations/components/generators/anthropic/generator.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Callable, ClassVar, Dict, List, Optional, Union
+
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.dataclasses import StreamingChunk
+from haystack.utils import Secret, deserialize_callable, deserialize_secrets_inplace, serialize_callable
+
+from anthropic import Anthropic, Stream
+from anthropic.types import (
+    ContentBlockDeltaEvent,
+    Message,
+    MessageDeltaEvent,
+    MessageParam,
+    MessageStartEvent,
+    MessageStreamEvent,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class AnthropicGenerator:
+    """
+    Enables text generation using Anthropic large language models (LLMs). It supports the Claude family of models.
+
+    Key Features and Compatibility:
+    - Primary Compatibility: Designed to work seamlessly with the Claude family of models.
+    - Streaming Support: Supports streaming responses from the Anthropic API.
+    - Customizability: Supports all parameters supported by the Anthropic API.
+
+    Input and Output Format:
+    - String Format: This component uses strings for both input and output.
+
+    ```python
+    from haystack_integrations.components.generators.anthropic import AnthropicGenerator
+
+    client = AnthropicGenerator(model="claude-2.1")
+    response = client.run("What's Natural Language Processing? Be brief.")
+    print(response)
+    >>{'replies': ['Natural language processing (NLP) is a branch of artificial intelligence focused on enabling
+    >>computers to understand, interpret, and manipulate human language. The goal of NLP is to read, decipher,
+    >> understand, and make sense of the human languages in a manner that is valuable.'], 'meta': {'model':
+    >> 'claude-2.1', 'index': 0, 'finish_reason': 'end_turn', 'usage': {'input_tokens': 18, 'output_tokens': 58}}}
+    ```
+    """
+
+    # The parameters that can be passed to the Anthropic API https://docs.anthropic.com/claude/reference/messages_post
+    ALLOWED_PARAMS: ClassVar[List[str]] = [
+        "system",
+        "max_tokens",
+        "metadata",
+        # "stream", explicitly passed to the interface
+        "stop_sequences",
+        "temperature",
+        "top_p",
+        "top_k",
+    ]
+
+    def __init__(
+        self,
+        api_key: Secret = Secret.from_env_var("ANTHROPIC_API_KEY"),  # noqa: B008
+        model: str = "claude-3-sonnet-20240229",
```

Comment:
As for the ChatGenerator: why not use claude-3-sonnet instead of claude-3-sonnet-20240229? Otherwise the default model will probably be outdated very quickly. Let's use claude-3-sonnet instead.

Reply:
```anthropic.NotFoundError: Error code: 404 - {'type': 'error', 'error': {'type': 'not_found_error', 'message': 'model: claude-3-sonnet'}}``` 😓